### PR TITLE
Haptic feedback generalized - DRV2605 and solenoids

### PR DESCRIPTION
**Pull request by ishtob**
- [x] Decoupled DRV2605L from the Hadron keymaps
- [x]  implement eeprom for DRV2605L configs
- [x] Decoupled solenoid from @mtdjr userspace
- [x] Implement eeprom for solenoid configs
- [x] Add haptic keycodes to quantum keycodes
- [x] write documentation

## Types of changes

- [x] Core
- [ ] Bugfix
- [x] New Feature
- [x] Enhancement/Optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/Layout/Userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by this PR

* 

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document. (https://docs.qmk.fm/#/contributing)
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/quantum_keycodes.h [hunk 1]
```diff
@@ -446,25 +446,35 @@ enum quantum_keycodes {
...
+
+        UNICODE_MODE_OSX,
+        UNICODE_MODE_LNX,
+        UNICODE_MODE_WIN,
+        UNICODE_MODE_BSD,
+        UNICODE_MODE_WINC,
+
+        HAPT_ON,
+        HAPT_OFF,
+        HAPT_TOGGLE,
```
Comment (mechmerlin) [inline quantum/quantum_keycodes.h]: for Backlights we have TOGG, for RGB we have TOG, and now we have TOGGLE. LOL. I think we should try and stick with one spelling of toggle.

File: quantum/quantum_keycodes.h [hunk 2]
```diff
@@ -446,25 +446,35 @@ enum quantum_keycodes {
...
+        HAPT_RESET,
+        HAPT_FEEDBACK,
+        HAPT_MODE_INC,
```
Comment (mechmerlin) [inline quantum/quantum_keycodes.h]: Same issue with these next 4. For Backlight we do explicitly do _INC and _DEC, but its just the I or D for RGB.

File: quantum/quantum_keycodes.h (hunk 1 above)
Comment (ishtob) [reply quantum/quantum_keycodes.h]: I have changed the spelling to fit the 7 charater paradigm, so they are all not HPT_*

Comment (drashna) [reply quantum/quantum_keycodes.h]: Well, you can have both.  If you add a define below, .... that's how many keycodes are operating, some of mine included

Comment (ishtob) [reply quantum/quantum_keycodes.h]: I think this may be fine for now without both, I've made a table in the docs to explain the keycodes anyways.

**Comment by ishtob** [comment]
think now we just need someone with a solenoid to make sure the code works in an actual setup

**Comment by ishtob** [comment]
tested working as of last night. This PR is ready for merge, unless it still needs to be reviewed by the collaborators

**Comment by mechmerlin** [comment]
Thanks ishtob, haptic feedback is pretty cool!

**Pull request merged**